web-src/src/components/ApplicationProvider.js

```diff
@@ -36,54 +37,67 @@ export const ApplicationContext = React.createContext(undefined);
 export const ApplicationProvider = ({ children }) => {
   const { user, done } = useShellContext();
   const setCustomPromptTemplates = useSetRecoilState(customPromptTemplatesState);
-  const [application, setApplication] = useState(undefined);
+  const [applicationContext, setApplicationContext] = useState(undefined);
 
   useEffect(() => {
     if (!user) {
       return;
     }
 
-    setApplication({
-      appVersion: APP_VERSION,
-
-      firefallService: new FirefallService({
-        completeEndpoint: actions[COMPLETE_ACTION],
-        feedbackEndpoint: actions[FEEDBACK_ACTION],
-        imsOrg: user.imsOrg,
-        accessToken: user.imsToken,
-      }),
-
-      csvParserService: new CsvParserService({
-        csvParserEndpoint: actions[CSV_PARSER_ACTION],
-      }),
-
-      targetService: new TargetService({
-        targetEndpoint: actions[TARGET_ACTION],
-        imsTenant: user.imsTenant,
-        accessToken: user.imsToken,
-      }),
-
-      expressSdkService: new ExpressSdkService({
-        clientId: 'aem-genai-assistant',
-        appName: 'AEM Generate Variations',
-        userId: user.id,
-        accessToken: user.imsToken,
-      }),
-    });
-
+    console.debug('Reading custom prompt templates...');
     readCustomPromptTemplates().then((templates) => {
       setCustomPromptTemplates(templates);
     });
 
-    done();
-  }, [user, done, setApplication]);
+    const createApplicationContext = async () => {
+      console.debug('Creating application context...');
+
+      const featureFlags = await getFeatureFlags();
+      console.debug('Feature flags:', featureFlags);
+
+      return {
+        appVersion: APP_VERSION,
+
+        featureFlags,
```

Comment:
for consistency, what about..?

```
featureFlagService: new FeatureFlagService({
   projectId: 'aem-generate-variations'
}),
```

where `featureFlagService` exposes  a get method: `featureFlagService.get('SITES-1234')` which returns a boolean.

Reply:
Agree with @basecode